### ISSUE TITLE
Add custom label to all objects in kubernetes chart

### ISSUE THREAD

**Issue by TheCubicleJockey** [projectcalico/calico#8391]
## Expected Behavior

This chart improvement would allow the user to include a defined label in the value.yaml, and that label would be included in the kubernetes objects in metadata.label where allowed by kubernetes.

## Current Behavior

Nothing like this is allowed in this chart.

## Possible Solution

Anything that allows user defined labels in the kubernetes objects. Using something like commonLabels at the top level of the values file.

## Steps to Reproduce (for bugs)

1. N/A

## Context

Labels are good in kubernetes for organization and for some custom tools that read labels as a part of rule sets.

## Your Environment

* Calico version: 3.26.3
* Orchestrator version: Kubernetes 1.28
* Operating System and version: Kubernetes 1.28
* Link to your project (optional):

**Comment by TheCubicleJockey** [comment]
Am I good to make the PR?

**Comment by caseydavenport** [comment]
@TheCubicleJockey right now, there is support for `podLabels`: https://github.com/projectcalico/calico/blob/master/charts/tigera-operator/values.yaml#L49-L50

Would you propose a new field that is merged with `podLabels`?

**Comment by TheCubicleJockey** [comment]
My use case is for the clusterrole to have a label. So, I could make one specifically for the clusterrole, but I figured most charts have a `values.commonLabels` or a `values.additionalLabels` as a more generally acceptable version that would fit most need and my needs. 

If i am doing the work, i would like to help improve the chart in a way that you or your team might like it improved.

**Comment by caseydavenport** [comment]
Yes, I think `values.additionalLabels` seems like a a good way to spin this, SGTM!

**Comment by TheCubicleJockey** [comment]
Made a [PR here ](https://github.com/projectcalico/calico/pull/8437)

**Comment by caseydavenport** [comment]
https://github.com/projectcalico/calico/pull/8722
